projects/plugins/jetpack/modules/publicize/enhanced-open-graph.php

```diff
@@ -16,6 +16,11 @@ function enhanced_og_image( $tags ) {
 
 	global $post;
 
+	// Bail if we do not have info about the post.
+	if ( empty( $post ) || ! is_a( $post, 'WP_Post' ) ) {
```

Comment:
No need for `empty()`, due to the `global $post` the variable will always exist. Also `instanceof` should do the right thing, it returns false for non-objects (and as far as I know the only reason to use `is_a()` over `instanceof` is if you're passing true for its `$allow_string` parameter).
```suggestion
	if ( ! $post instanceof WP_Post ) {
```
(same below)